### ISSUE TITLE
No longer working for me

### ISSUE THREAD

**Issue by User2020** [timur-tabi/amazon-vine#2]
I have tried a few agent strings and it is always returning 0 items.  Anyone else?  Even using the agent string of my current browser which works fine.

**Comment by timur-tabi** [comment]
Amazon might be clamping down on robots.  I will have to add code that checks for that.  In the meantime, I suggest you stop using the script.

**Comment by User2020** [comment]
Already done.  Please let me know what you find.  Thanks so much.

**Comment by timur-tabi** [comment]
So I just tried it again after waiting a few days, and it works now.  Perhaps Amazon doesn't like the 10 minute waits.  Try increasing it to 30 minutes ("-w 30") and see if it's more reliable.

I'll try to add code to reduce the number of repeat logins, but I had a hard time getting that to work last time I tried to implement that.

**Comment by User2020** [comment]
I wonder if it is being blocked on a per-user basis?  Mine still isn't
working.  I had previously modified mine to be more than 10 minutes
already, so I wonder what the deal is.  I miss having the script working
 :(  lol

**Comment by timur-tabi** [comment]
Are you sure you're using the latest version?  The latest version includes a user agent randomizer.

**Comment by User2020** [comment]
I am using a slightly customized version and haven't integrated the
randomizer in there yet.  But I changed my user agent string to be the same
as my browser, which Vine works fine in.  Shouldn't that work?

**Comment by User2020** [comment]
Ok, just traded my custom version with my browser's user agent string out
for your script and it still is reporting 0 items every time.  Bummer.

**Comment by timur-tabi** [comment]
Can you send me the output of debug.html and debug.txt?

**Comment by User2020** [comment]
AH-HA!  Debug.txt is empty but take a look at the HTML.  I removed my
e-mail address from the source code (it was in the e-mail address field).
But take a look at what they're doing.  (My password hasn't changed and is
correct.)

Thanks for looking into this!  I really appreciate your work.

**Comment by User2020** [comment]
I should note that I have never once seen a CAPTCHA when logging in from my
browsers.

**Comment by timur-tabi** [comment]
I don't see the debug.html from you, but if you get a captcha, like "<img alt="captcha" src="...", then Amazon thinks you're a robot because you've logged in too many times to that account from that IP address.  Any chance you can run this script from a different network, like from a public wifi?

I'm modifying the script to detect captchas, but I don't think I'll ever be able to handle them.

**Comment by User2020** [comment]
Here ya go.  Didn't get attached when sent via e-mail.

I will reset my IP by resetting my modem like I did the other day again... didn't help the other day, but will try again. Thanks
[debug.html.TXT](https://github.com/timur-tabi/amazon-vine/files/46143/debug.html.TXT)

**Comment by User2020** [comment]
Funny thing is that I don't check Vine for All, only my queue.  And I have it set for longer than 10 minutes and it still hates me  LOL

**Comment by timur-tabi** [comment]
Can you try this version: http://pastebin.com/AqHiXtsM

It will open a browser window and show the captca image, then prompt you to type in the word.  I think the word is too hard to read, so you may need to relaunch the script a couple times until it works.

**Comment by User2020** [comment]
I get a response.html page now and it is trying to verify my identity.  I
tried opening the HTML page and entering the data but it doesn't submit
properly:

We haven't seen you using this device before. To help protect your account,
we just want to make sure it's really you.
To continue, please answer one of the security questions below
1. What is the ZIP code of the billing address you registered with your
credit or debit card number that ends in XX?

Please enter a complete 5-digit ZIP Code.
OR
2. What is your phone number that ends in XX?

Please enter a complete 10-digit phone number, including the area code.
Sign in using our secure server

**Comment by timur-tabi** [comment]
Well, I've never seen that before.  I don't understand why you're able to log in via a web browser, but you still have these problems with my script.  Are you using a proxy with your web browser?

**Comment by User2020** [comment]
No.  Nothing special as far as setup, and nothing changed.  Was working up
until a few days ago and then noticed it was reporting 0 items all the
time.  It is running on a Pi, BTW.  I signed in using the Pi browser and it
is fine.  So are Windows browsers.  Will try to run script on Windows
machine next instead of Pi.

**Comment by timur-tabi** [comment]
What exactly is the user agent string on your Pi when using the Pi browser?

**Comment by User2020** [comment]
Pi Chromium browser (the last version ever made):  Mozilla/5.0 (X11; Linux
armv6l) AppleWebKit/537.4 (KHTML, like Gecko) Chrome/22.0.1229.94
Safari/537.4

But my Windows string is the string i always used.  Neither works.

Also tried the randomizer.  Apparently keeps going to the verification
screen no matter what.  I will try running Python on my Windows machine
tomorrow to see if that helps.  I really would like to keep the Pi running
though but if I can't anymore, I guess I don't have a choice.  So weird!

**Comment by User2020** [comment]
Well, that didn't help.  Ran the script on my Windows machine and again is
resulted in a debug page "We haven't seen you using this device before. To
help protect your account, we just want to make sure it's really you."

**Comment by timur-tabi** [comment]
There may be some other http header that Amazon is expecting or doesn't trust.  Somehow, Amazon has identified the script.  Can you try playing with br.addheaders to add more headers that your web browser is using?  You could use Firefox's debugging feature to display the http packets being sent by your browser.

**Comment by User2020** [comment]
Is there a way to send you a file privately?  I'd like to send you the
complete debug.html that asks for my info.  Maybe you can think of a magic
way to define the data in the code and input the data automatically if that
screen is detected?  But I'd rather not post it here.

**Comment by timur-tabi** [comment]
timur at tabi dot org

**Comment by timur-tabi** [comment]
This should be fixed now.
